src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,324 @@
+                {
+                    ["profileKey"] = request.QualityProfileKey,
+                    ["ps"] = request.PageSize.ToString()
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => ProcessJsonResponse<QualityProfileChangeLogResponse>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileResponse[]>> GetQualityProfilesAsync(QualityProfileRequest request,
+            CancellationToken token)
+        {
+            var queryParams = new Dictionary<string, string>();
+            if (request.ProjectKey == null)
+            {
+                queryParams["defaults"] = "true";
+            }
+            else
+            {
+                queryParams["projectKey"] = request.ProjectKey;
+            }
+
+            // Since 5.2
+            var apiPath = BuildRelativeUrl("api/qualityprofiles/search", queryParams);
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => JObject.Parse(stringResponse)["profiles"].ToObject<QualityProfileResponse[]>());
+        }
+
+        public Task<Result<RoslynExportProfileResponse>> GetRoslynExportProfileAsync(RoslynExportProfileRequest request,
+            CancellationToken token)
+        {
+            // Since 5.2
+            var apiPath = BuildRelativeUrl("api/qualityprofiles/export",
+                new Dictionary<string, string>
+                {
+                    ["name"] = request.QualityProfileName,
+                    ["language"] = request.LanguageKey,
+                    ["exporterKey"] = string.Format(CultureInfo.InvariantCulture, "roslyn-{0}", request.LanguageKey)
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse =>
+                {
+                    using (var reader = new StringReader(stringResponse))
+                    {
+                        return RoslynExportProfileResponse.Load(reader);
+                    }
+                });
+        }
+
+        public Task<Result<VersionResponse>> GetVersionAsync(CancellationToken token)
+        {
+            // Since 2.10; internal
+            var serverVersionAPI = BuildRelativeUrl("api/server/version");
+
+            return InvokeSonarQubeApi(serverVersionAPI, token,
+                stringResponse => new VersionResponse { Version = stringResponse });
+        }
+
+        public Task<Result<CredentialResponse>> ValidateCredentialsAsync(CancellationToken token)
+        {
+            // Since 3.3
+            var validateCredentialsApiUrl = BuildRelativeUrl("api/authentication/validate");
+
+            return InvokeSonarQubeApi(validateCredentialsApiUrl, token,
+                stringResponse => new CredentialResponse { IsValid = (bool)JObject.Parse(stringResponse).SelectToken("valid") });
+        }
+
+        private Uri BuildRelativeUrl(string relativePath, Dictionary<string, string> queryParameters = null)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>>' instead of 'System.Collections.Generic.Dictionary<string, string>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)